src/dcap_provider.cpp

```diff
@@ -861,24 +1084,82 @@ static quote3_error_t convert_to_intel_error(sgx_plat_error_t platformError)
 
 static std::string build_pck_crl_url(
     std::string crl_name,
-    std::string api_version)
+    std::string api_version,
+    sgx_prod_type_t prod_type = SGX_PROD_TYPE_SGX)
 {
-    std::string version = get_collateral_version();
+    std::string version;
+    std::stringstream url;
+    std::string client_id;
+    if (prod_type == SGX_PROD_TYPE_TDX)
+    {
+        version = get_collateral_version_tdx();
+
+        client_id = get_client_id();
+
+        url << get_base_url_tdx();
+
+        if (!version.empty())

```

Comment:
if version is empty, throw an error here

Reply:
If you are making this change please make it everywhere

Reply:
Looking at the code I don't see why this check was being made, since get_collateral_version() never returns an empty string. Same with client id. I'll remove it.